impeller/renderer/backend/gles/reactor_gles.h

```diff
@@ -16,6 +16,14 @@
 
 namespace impeller {
 
+/// @brief Storage for either a GL handle or sync fence.
+struct GLHandle {
+  union {
+    GLuint handle;
+    GLsync sync;
+  };
+};
+
```

Comment:
This seems error prone to have a union that doesn't store the type.  It is stored in the HandleGLES.  Would it make more sense to have the enum there?  In fact it's pretty confusing to introduce a GLHandle when there is already the HandleGLES.

Reply:
I can make it more obvious, but this type is essentially private to the reactor and the storage is an implementation detail. The reactor is capabale of maintaining the invariants associated with the union anyway

Reply:
Now called GLStorage and private to the reactor.

Reply:
That's an improvement, thanks.  Why don't we just have a different set of functions though for sync versus handle?  Then we wouldn't need the union type at all.  The HandleGLES will know by his type which functions to call.

Reply:
Ahh, it's complicated by our storage of the LiveHandles in `handles_`.  I think we could have sorted this out better but addressing it is outside the scope of this PR.  For future reference, I don't like the struct with the union because it's correct usage is enforced by logic that exists outside of it.  We could have shuffled things around to avoid that, but this lgtm for now.